src/VisualStudio/TestUtilities/IntegrationHost.cs

```diff
@@ -0,0 +1,522 @@
+            var monikers = new IMoniker[1];
+
+            Ole32.GetRunningObjectTable(0, out runningObjectTable);
+            runningObjectTable.EnumRunning(out enumMoniker);
+            Ole32.CreateBindCtx(0, out bindContext);
+
+            do
+            {
+                monikers[0] = null;
+
+                var monikersFetched = 0u;
+                var result = enumMoniker.Next(1, monikers, out monikersFetched);
+
+                if (result != 0)
+                {
+                    return null;
+                }
+
+                var moniker = monikers[0];
+                var fullDisplayName = null as string;
+
+                moniker.GetDisplayName(bindContext, null, out fullDisplayName);
+
+                var displayNameProcessId = 0;
+
+                if (!int.TryParse(fullDisplayName.Split(':').Last(), out displayNameProcessId))
+                {
+                    continue;
+                }
+
+                var displayName = fullDisplayName.Substring(0, (fullDisplayName.Length - (displayNameProcessId.ToString().Length + 1)));
+                var fullProgId = VsProgId.StartsWith("!") ? VsProgId : $"!{VsProgId}";
+
+                if (displayName.Equals(fullProgId, StringComparison.OrdinalIgnoreCase))
+                {
+                    if (displayNameProcessId == process.Id)
+                    {
+
+                        runningObjectTable.GetObject(moniker, out dte);
+                    }
+                }
+            }
+            while (dte == null);
+
+            return (DTE)(dte);
+        }
+
+        public void Cleanup()
+        {
+            // Clean the Backup Files folder so we avoid the recovery dialog on launch.
+            IntegrationHelper.DeleteDirectoryRecursively(Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments), VsUserFilesFolder, "Backup Files"));
+
+            CleanupDte();
+
+            if (RequireNewInstance)
+            {
+                IntegrationLog.Current.WriteLine("Closing existing Visual Studio instance.");
+
+                CleanupTimeoutWatcherThread();
+                CleanupRemotingService();
+                CleanupModalWatcherThread();
+                CleanupHostProcess();
+            }
+        }
+
+        public void ClickAutomationElement(string elementName, bool recursive = false, int timeout = IntegrationHelper.DefaultTimeout)
+        {
+            var automationElement = LocateAutomationElement(elementName, recursive, timeout);
+
+            var invokePattern = null as object;
+            if (automationElement.TryGetCurrentPattern(InvokePattern.Pattern, out invokePattern))
+            {
+                ((InvokePattern)(invokePattern)).Invoke();
+            }
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+            // We explicitly don't call GC.SuppressFinalize() so that
+            // the finalizer is still called and the _hostProcess is killed
```

Comment:
Why is it important that `_hostProcess` is killed in the finalizer vs. right now?